vote.py

```diff
@@ -0,0 +1,181 @@
+from brutal.core.plugin import BotPlugin, cmd
+from brutal.core.utils import split_args_by
+
+
+class Vote(BotPlugin):
+    VOTING_NONE = 0
+    VOTING_YESNO = 1
+    VOTING_QUESTION = 2
+
+    def setup(self, *args, **kwargs):
+        self.reset()
+
+    def reset(self):
+        """Reset all voting variables"""
+        self.voting = self.VOTING_NONE
+        self.yeses = 0
+        self.nos = 0
+        self.voted = {}
+
+    def voting_in_progress(self, question=False):
+        if question:
+            return self.voting == self.VOTING_QUESTION
+        return self.voting is not self.VOTING_NONE
+
+    def judgement(self, event=None, penalty=''):
+        """Summarize votes and make decision"""
+        # TODO: if voting succeeded, get op and punish user
+        msg = 'unsuccessfully'
+        if self.yeses + self.nos > 1 and self.yeses > self.nos:
+            msg = 'successfully '
+
+        msg += 'with {0}x yes, {1}x no'.format(self.yeses, self.nos)
+        result = '{0} {1}'.format(penalty, event.args[0])
+
+        self.reset()
+
+        return 'Voting to {0} finished {1}'.format(result, msg)
+
+    def vote_result(self, event=None, penalty=''):
+        """Start voting of given penalty"""
+        if self.voting_in_progress() or self.voting_in_progress(question=True):
+            return 'Voting in progress...'
+
+        if not len(event.args):
+            return 'No user specified'
+
+        user = event.args[0]
+        self.voting = self.VOTING_YESNO
+
+        self.delay_task(60, self.judgement, event=event, penalty=penalty)
```

Comment:
As a channel operator I'd like to have the ability to cast a quick vote, or even some longer ones (say for 5 minutes).